homeassistant/components/wilight/switch.py

```diff
@@ -111,54 +111,48 @@ async def async_setup_entry(
     async_add_entities(entities)
 
     # Handle services for a discovered WiLight device.
-    async def set_watering_time(service: Any) -> None:
+    async def set_watering_time(entity, service: Any) -> None:
         entity_id = service.data[ATTR_ENTITY_ID]
 
-        for entity in entities:
-            if (entity.entity_id == entity_id) & isinstance(entity, WiLightValveSwitch):
-                watering_time = service.data[ATTR_WATERING_TIME]
-                await entity.async_set_switch_time(watering_time=watering_time)
+        if (entity.entity_id == entity_id) & isinstance(entity, WiLightValveSwitch):
```

Comment:
Raise a `ValueError` with a message as argument here if the entity_id targets the wrong kind of device. Ie invert the check, raise if true and then we can outdent below.

Reply:
Done